gateway/message_handler.go

```diff
@@ -193,11 +158,92 @@ func (sc *SlackClient) handleMessageEvent(incomingChan chan<- *SlackEvent, messa
 		return
 
 	case "message_replied":
-		// As far as I can tell, this is only useful for updating the reply count of a message thread.
-		return
+		// Handle thread replies by maybe quoting the parent message if enough time has passed.
+		subMessage := messageData.SubMessage
+		if subMessage == nil {
+			log.Printf("%s message_replied with missing submessage: %+v SubMessage:%+v",
+				sc.Tag(), messageData, messageData.SubMessage)
+			return
+		}
+
+		if subMessage.Type != "message" {
+			log.Printf("%s message_replied with unexpected submessage type: %+v SubMessage:%+v",
+				sc.Tag(), messageData, messageData.SubMessage)
+			return
+		}
+
+		if !sc.shouldQuoteThreadParent(subMessage.ThreadTimestamp, messageData.Timestamp) {
+			return
+		}
+
+		sender, _, wasSenderSwapped = sc.resolveMessageEndpoints(subMessage)
+		if sender == nil || target == "" {
+			return
+		}
+
+		messageText := sc.ParseMessageText(subMessage.Text)
+		for _, attachment := range subMessage.Attachments {
```

Comment:
If we're overwriting `messageText` each time, why don't we just grab the first or last attachment in `subMessage.Attachments` instead?